.eslintrc.js

```diff
@@ -1,63 +1,242 @@
+
+        // Increase the max line length to 120. The rest of this setting is copied from the AirBnB config.
+        'max-len': ['error', 120, 2, {
+          ignoreUrls: true,
+          ignoreComments: false,
+          ignoreRegExpLiterals: true,
+          ignoreStrings: true,
+          ignoreTemplateLiterals: true,
+        }],
+
+        // Restrict the use of backticks to declare a normal string. Template literals should only be used when the
+        // template string contains placeholders. The rest of this setting is copied from the AirBnb config.
+        quotes: ['error', 'single', { avoidEscape: true, allowTemplateLiterals: false }],
+
+        // TODO: not sure if this will work because of plugin loading on .js files
+        // Allow leading underscores for parameter names, which is used to acknowledge unused variables in TypeScript.
+        // Also, enforce camelCase naming for variables. Ideally, the leading underscore could be restricted to only
+        // unused parameter names, but this rule isn't capable of knowing when a variable is unused. The camelcase and
+        // no-underscore-dangle rules are replaced with the naming-convention rule because this single rule can serve
+        // both purposes, and it works fine on non-TypeScript code.
+        camelcase: 'off',
+        'no-underscore-dangle': 'off',
+        '@typescript-eslint/naming-convention': [
+          'error',
+          {
+            selector: 'default',
+            format: ['camelCase'],
+          },
+          {
+            selector: 'variable',
+            // PascalCase for variables is added to allow exporting a singleton, function library, or bare object as in
+            // section 23.8 of the AirBnB style guide
+            format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+          },
+          {
+            selector: 'parameter',
+            format: ['camelCase'],
+            leadingUnderscore: 'allow',
+          },
+          {
+            selector: 'typeLike',
+            format: ['PascalCase'],
+          },
+        ],
+
+        // Allow cyclical imports. Turning this rule on is mainly a way to manage the performance concern for linting
+        // time. Our projects are not large enough to warrant this. Overrides AirBnB styles.
+        'import/no-cycle': 'off',
+
+        // Prevent importing submodules of other modules. Using the internal structure of a module exposes
+        // implementation details that can potentially change in breaking ways. Overrides AirBnB styles.
+        'import/no-internal-modules': ['error', {
+          // Use the following option to set a list of allowable globs in this project.
+          allow: [
+            '**/middleware/*', // the src/middleware directory doesn't export a module, it's just a namespace.
+          ],
+        }],
+
+        // Remove the minProperties option for enforcing line breaks between braces. The AirBnB config sets this to 4,
```

Comment:
Assuming I'm understanding this description correctly, I am a _huge_ fan of this change.